src/status_im/extensions/capacities/components.cljs

```diff
@@ -25,26 +28,44 @@
     (js/clearTimeout id))
   (reset! current (js/setTimeout #(on-input-change-text on-change value) delay)))
 
-(defn input [{:keys [keyboard-type style error on-change change-delay placeholder placeholder-text-color selection-color
+(defn- in-range
+  ;; Determines whether a number is in a range
+  [n, start, end]
+  (cond
+    (nil? n) true
+    (and (nil? start) (nil? end)) true
+    (= n "") true
+    (and (< n end) (> n start)) true
+    :else false))
+
+(defn input [{:keys [keyboard-type style error min max on-change change-delay placeholder placeholder-text-color selection-color
                      auto-focus on-submit default-value]}]
-  [react/view
-   [react/text-input (merge {:placeholder placeholder}
-                            (when placeholder-text-color {:placeholder-text-color placeholder-text-color})
-                            (when selection-color {:selection-color selection-color})
-                            (when style {:style style})
-                            (when keyboard-type {:keyboard-type keyboard-type})
-                            (when auto-focus {:auto-focus auto-focus})
-                            (when default-value {:default-value default-value})
-                            (when on-submit
-                              {:on-submit-editing #(on-submit {})})
-                            (when on-change
-                              {:on-change-text
-                               (if change-delay
-                                 (let [current (atom nil)]
-                                   #(on-input-change-text-delay current on-change % change-delay))
-                                 #(on-input-change-text on-change %))}))]
-   (when error
-     [tooltip/tooltip error (styles/error error)])])
+
+  (let [input-range (reagent/atom nil)]
+    (fn []
```

Comment:
Why the fn here?

Reply:
@jeluard I tried to do in different ways, and only works when there's a anonymous function to handle. In some tutorials about atoms and event handling using clojure, reagent and react it was using this approach.

Reply:
What didn't work w/o it?

Reply:
@jeluard I think it is related to event handling with reagent, it needs to re-render once the atom change with the updated values.

You can see here:
https://github.com/status-im/status-react/blob/develop/src/status_im/ui/screens/wallet/components/views.cljs#L229

It is a similar situation where `fn` is used, I there's many of those inside the project.

Reply:
But what didn't work exactly? Not clear from your comments

Reply:
Sorry @jeluard, without `fn` wrapped on the component when using `reagent/atom`, the input don't update the atom value and didn't check on `on-change` for the `in-range` function, so nothing happens if I get rid of the `fn` inside. This means the validation and the input error don't appear and nothing happens.